tools/kagami/src/main.rs

```diff
@@ -774,3 +777,114 @@ mod tokens {
         }
     }
 }
+
+mod examples {
+    use clap::{Parser, Subcommand};
+    use color_eyre::Result;
+    use iroha_crypto::{KeyPair, PublicKey};
+
+    use super::*;
+
+    #[derive(Parser, Debug, Clone)]
+    pub struct Args {
+        #[clap(subcommand)]
+        usecase: Usecase,
+    }
+
+    impl<T: Write> RunArgs<T> for Args {
+        fn run(self, writer: &mut BufWriter<T>) -> Outcome {
+            let instructions = match self.usecase {
+                Usecase::CreateAccount {
+                    account_id,
+                    public_key,
+                } => create_account(account_id, public_key)?,
+                Usecase::MintPublicKey {
+                    account_id,
+                    public_key,
+                } => mint_public_key(account_id, public_key)?,
+                Usecase::BurnPublicKey {
+                    account_id,
+                    public_key,
+                } => burn_public_key(account_id.as_str(), public_key.as_str())?,
+            };
+            write!(
+                writer,
+                "{}",
+                serde_json::to_string_pretty(&instructions).wrap_err("Serialization error")?
+            )
+            .wrap_err("Failed to write.")
+        }
+    }
+
+    #[derive(Subcommand, Debug, Clone)]
+    pub enum Usecase {
+        /// Generate an instruction that creates a new account.
+        CreateAccount {
+            /// A new account id.
+            #[clap(long)]
+            account_id: Option<String>,
+            /// A new account public key.
+            /// If this flag isn't provided, will generate a new key pair.
```

Comment:
It is an argument/option rather than a flag. Booleans are called flags.

```suggestion
            /// If this argument isn't provided, will generate a new key pair.
```

It is also relevant for further doc comments.